Source/Events.Processing/Filters/FilterValidators.cs

```diff
@@ -2,50 +2,99 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Dolittle.Collections;
 using Dolittle.DependencyInversion;
+using Dolittle.Lifecycle;
 using Dolittle.Logging;
+using Dolittle.Types;
 
 namespace Dolittle.Runtime.Events.Processing.Filters
 {
     /// <summary>
     /// Represents an implementation of <see cref="IFilterValidators" />.
     /// </summary>
+    [Singleton]
     public class FilterValidators : IFilterValidators
     {
+        readonly ITypeFinder _typeFinder;
         readonly IContainer _container;
         readonly ILogger _logger;
+        readonly IDictionary<Type, Type> _filterDefinitionToValidatorMap = new Dictionary<Type, Type>();
 
         /// <summary>
         /// Initializes a new instance of the <see cref="FilterValidators"/> class.
         /// </summary>
+        /// <param name="typeFinder">The <see cref="ITypeFinder" />.</param>
         /// <param name="container">The <see cref="IContainer" />.</param>
         /// <param name="logger">The <see cref="ILogger" />.</param>
-        public FilterValidators(IContainer container, ILogger logger)
+        public FilterValidators(ITypeFinder typeFinder, IContainer container, ILogger logger)
         {
+            _typeFinder = typeFinder;
             _container = container;
             _logger = logger;
+            PopulateFilterValidatorMap();
```

Comment:
Instead of doing this we could have an ICanProvideBindings which binds them up using the TypeFinder.